### PR TITLE
[kf5codecs] add explicit gperf dependency

### PR DESCRIPTION
**Pull request by wrobelda**
**Describe the pull request**

- #### What does your PR fix?  
  This prevents built failures on macOS with its outdated `GNU gperf 3.0.3` that chokes on kf5codec's 'register' specifiers.
  Also wraps paths in quotes and removes the line deleting `/etc`, which isn't being populated.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  No change.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  No change.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/kf5codecs/portfile.cmake [hunk 1]
```diff
@@ -28,9 +24,8 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
     file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
 endif()
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/bin/data)
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/etc)
```
Comment (JackBoosY) [inline ports/kf5codecs/portfile.cmake:32]: Why do you remove this line? Does _etc_ not exist anymore?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/kf5codecs/portfile.cmake (hunk 1 above)
Comment (wrobelda) [reply ports/kf5codecs/portfile.cmake:32]: "Also wraps paths in quotes and removes the line deleting /etc, which isn't being populated." – the `/etc/` was never populated, that line was a copy and paste from another port file.

**Review by JackBoosY — APPROVED** [review]

File: ports/kf5codecs/vcpkg.json [hunk 2]
```diff
@@ -1,11 +1,12 @@
 {
   "name": "kf5codecs",
   "version": "5.84.0",
-  "port-version": 2,
+  "port-version": 3,
   "description": "String encoding library",
   "homepage": "https://api.kde.org/frameworks/kcodecs/html/index.html",
   "dependencies": [
     "ecm",
+    "gperf",
```
Comment (BillyONeal) [inline ports/kf5codecs/vcpkg.json:9]:
```suggestion
    {
      "name": "gperf",
      "host": true
    }
```

It looks like this intends to run gperf so it should be a host dependency?

**Review by BillyONeal — CHANGES_REQUESTED** [review]

File: ports/kf5codecs/vcpkg.json (hunk 2 above)
Comment (wrobelda) [reply ports/kf5codecs/vcpkg.json:9]: Good call, thanks.

Comment (wrobelda) [reply ports/kf5codecs/vcpkg.json]: Well, now it fails with:
```
CMake Error at D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find Gperf (missing: Gperf_EXECUTABLE)
Call Stack (most recent call first):
  D:/downloads/tools/cmake-3.21.1-windows/cmake-3.21.1-windows-i386/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  D:/installed/x64-windows-static/share/ECM/find-modules/FindGperf.cmake:60 (find_package_handle_standard_args)
  C:/a/1/s/scripts/buildsystems/vcpkg.cmake:786 (_find_package)
  CMakeLists.txt:23 (find_package)
```

Comment (wrobelda) [reply ports/kf5codecs/vcpkg.json]: Maybe I should instead:
1) Bump win32 `gperf` version and add a `BREW_PACKAGE_NAME`: https://github.com/microsoft/vcpkg/blob/8275003fd8c5119cb9438fea1425ebe774f2e527/scripts/cmake/vcpkg_find_acquire_program.cmake#L333
2) Add a warning to kf5codec regarding minimum `gperf` version required

Comment (dg0yt) [reply ports/kf5codecs/vcpkg.json]: gperf installs these files:
~~~
gperf:/share/gperf/copyright
gperf:/share/gperf/vcpkg_abi_info.txt
gperf:/tools/gperf.exe
~~~
There is no `vcpkg-cmake-config.cmake`, so there is nothing magically injecting the host binary path to the `FindGperf.cmake` search process. (Do you remember `gettext[tools]`? Or #17607?)
But you can probaby preset the desired `Gperf_EXECUTABLE` cache variable in `portfile.cmake`.

Comment (wrobelda) [reply ports/kf5codecs/vcpkg.json]: @dg0yt I just checked on macOS here and it does find it just fine in `tools/gperf` without any extra steps, so this is something with Windows:

```
-- Found Gperf: /Users/cromo/Documents/Sourcecode/vcpkg/installed/arm64-osx/tools/gperf
```

Comment (dg0yt) [reply ports/kf5codecs/vcpkg.json]: On your mac, is host triplet = target triplet?
On a standard modern windows PC, the triplets differ.

**Comment by JackBoosY** [comment]
Please get failure logs [here](https://dev.azure.com/vcpkg/public/_build/results?buildId=60280&view=artifacts&pathAsName=false&type=publishedArtifacts).

**Comment by wrobelda** [comment]
@JackBoosY I addressed @BillyONeal concern, all issues are fixed now.

**Comment by JackBoosY** [comment]
@wrobelda Yes, we are handling this.

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**